pkg/conversion/conversion.go

```diff
@@ -2,12 +2,24 @@
 package conversion
 
 import (
+	"regexp"
+	"strings"
+	"unicode"
+
 	"github.com/netobserv/network-observability-operator/controllers/constants"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/util/json"
 )
 
+// Following K8S convention, mixed capitalization should be preserved
+// see https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#constants
+var upperPascalExceptions = map[string]string{
+	"IPFIX":        "IPFIX",
+	"EBPF":         "eBPF",
+	"SCRAM-SHA512": "ScramSHA512",
+}
```

Comment:
Managing exceptions manually here. eBPF and IPFIX are kept as their original acronyms case.

Following discussion with @jotak, he suggested to use `ScramSHA512` for better readability.
In AMQ-Streams, this is fully lowercase as `scram-sha-512` :disappointed: We may consider `SCRAM-SHA-512` as alternative.